addons/event_sale/models/account_move.py

```diff
@@ -10,7 +11,32 @@ class AccountMove(models.Model):
     def _invoice_paid_hook(self):
         """ When an invoice linked to a sales order selling registrations is
         paid confirm attendees. Attendees should indeed not be confirmed before
-        full payment. """
+        full payment. Post tickets on paid orders too, accessible from portal."""
         res = super(AccountMove, self)._invoice_paid_hook()
-        self.mapped('line_ids.sale_line_ids')._update_registrations(confirm=True, mark_as_paid=True)
+        sale_order_line_ids = self.mapped('line_ids.sale_line_ids')
+        sale_order_line_ids._update_registrations(confirm=True, mark_as_paid=True)
+
+        RegistrationSudo = self.env['event.registration'].sudo()
+        registrations_sudo = RegistrationSudo.search([('sale_order_line_id', 'in', sale_order_line_ids.ids)])
+        registrations_per_order_event = defaultdict(lambda: defaultdict(lambda: self.env['event.registration']))
+        for registration in registrations_sudo:
+            registrations_per_order_event[registration.sale_order_id][registration.event_id] += registration
+
+        for order, registrations_per_event in registrations_per_order_event.items():
+            pdfs = []
+            partner = order.partner_id
+            for event in registrations_per_event:
+                pdf = self.env['ir.actions.report'].sudo()._render_qweb_pdf('event.action_report_event_registration_full_page_ticket', registrations_per_event[event].ids)[0]
```

Comment:
is it possible to render in batch ? (and here you just get the value from the batched result)

Reply:
Oh sorry, I tough `_render_qweb_pdf` was returning a dict with many template but no, you can forget my message